src/jenkins.py

```diff
@@ -39,43 +45,43 @@ def wait_jenkins_ready(timeout: int = 140, check_interval: int = 10) -> None:
         TimeoutError: if Jenkins status check did not pass within the timeout duration.
     """
     start_time = datetime.now()
+    now = datetime.now()
     min_wait_seconds = timedelta(seconds=timeout)
-    while True:
+    while now - start_time < min_wait_seconds:
         if _is_jenkins_ready():
             break
         now = datetime.now()
-        if now - start_time > min_wait_seconds:
-            raise TimeoutError("Timed out waiting for Jenkins to become ready.")
         sleep(check_interval)
+    else:
+        raise TimeoutError("Timed out waiting for Jenkins to become ready.")
 
 
-def get_admin_credentials(password_file_contents: str) -> Credentials:
+def get_admin_credentials(connectable_container: Container) -> Credentials:
     """Retrieve admin credentials.
 
     Args:
-        password_file_contents: The contents inside `$JENKINS_HOME/secrets/initialAdminPassword`
-        file.
+        connectable_container: Connectable container to interact with filesystem.
 
     Returns:
         The Jenkins admin account credentials.
     """
     user = "admin"
```

Comment:
Is there any chance this could be changed by the operator?

Reply:
Not at the moment, it cannot be changed.